main.tf

```diff
@@ -0,0 +1,119 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+module "resource_names" {
+  source  = "d2lqlh14iel5k2.cloudfront.net/module_library/resource_name/launch"
+  version = "~> 1.0"
+
+  for_each = var.resource_names_map
+
+  region                  = join("", split("-", var.location))
+  class_env               = var.environment
+  cloud_resource_type     = each.value.name
+  instance_env            = var.environment_number
+  instance_resource       = var.resource_number
+  maximum_length          = each.value.max_length
+  logical_product_family  = var.logical_product_family
+  logical_product_service = var.logical_product_service
+  use_azure_region_abbr   = var.use_azure_region_abbr
+}
+
+module "resource_group" {
+  source  = "d2lqlh14iel5k2.cloudfront.net/module_primitive/resource_group/azurerm"
+  version = "~> 1.0"
+
+  name     = local.resource_group_name
+  location = var.location
+  tags     = local.resource_group_tags
+}
+
+module "key_vault" {
+  source  = "d2lqlh14iel5k2.cloudfront.net/module_primitive/key_vault/azurerm"
+  version = "~> 2.0"
+
+  resource_group = local.resource_group
+  key_vault_name = local.key_vault_name
+
+  enabled_for_deployment          = var.enabled_for_deployment
+  enabled_for_template_deployment = var.enabled_for_template_deployment
+  soft_delete_retention_days      = var.soft_delete_retention_days
+  purge_protection_enabled        = var.purge_protection_enabled
+  sku_name                        = var.sku_name
+  enable_rbac_authorization       = var.enable_rbac_authorization
+  public_network_access_enabled   = var.public_network_access_enabled
+
+  network_acls    = var.network_acls
+  access_policies = var.access_policies
+  custom_tags     = local.key_vault_tags
+
+  depends_on = [module.resource_group]
+}
+
+module "role_assignment" {
+  source  = "d2lqlh14iel5k2.cloudfront.net/module_primitive/role_assignment/azurerm"
+  version = "~> 1.0"
+
+  for_each = local.role_assignments
+
+  name                 = each.value.name
+  scope                = each.value.scope
+  role_definition_name = each.value.role_definition_name
+  principal_id         = each.value.principal_id
+
+  depends_on = [module.key_vault]
+}
+
+module "private_dns_zone" {
+  source  = "d2lqlh14iel5k2.cloudfront.net/module_primitive/private_dns_zone/azurerm"
+  version = "~> 1.0"
+
+  resource_group_name = local.resource_group_name
+  zone_name           = var.zone_name
+  soa_record          = var.soa_record
+  tags                = local.private_dns_zone_tags
+
+  depends_on = [module.resource_group]
+}
+
+module "private_dns_zone_link_vnet" {
```

Comment:
In addition to having vnet link for the vnet in which the private endpoint is created, its also a good idea to pass in a variable of additional vnet ids for which you need to create vnet links. Because in case of hub-and-spoke architecture where this key vault will be referenced from other vnets, the vnet-link must be created for each vnet. Something similar i have implemented here https://github.com/launchbynttdata/tf-azurerm-module_reference-kubernetes_cluster/blob/main/variables.tf#L136